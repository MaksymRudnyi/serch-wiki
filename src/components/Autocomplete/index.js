import React from 'react';
import ReactAutocomplete from 'react-autocomplete'

import Input from '../Input';

const Autocomplete = ({ articles, searchValue, onSearchChange }) => (
    <ReactAutocomplete
        items={articles}
        renderInput={Input}
        inputProps={{placeholder: 'Input a search term: '}}
        getItemValue={item => item.label}
        renderMenu={(children, value, style) => {
            return articles && articles.length ?
                (<div style={{...style}} className="input-suggestions">
                        {children}
                        <a href={`/search?query=${value}`} className="search-link">
                            See all results
                        </a>
                    </div>
                ) : <></>;
        }}
        renderItem={(item, highlighted) =>
            <div
                key={item.id}
                style={{backgroundColor: highlighted ? '#eee' : 'transparent'}}
            >
                {item.label}
            </div>
        }
        value={searchValue}
        onChange={onSearchChange}
    />
);

export default Autocomplete;
