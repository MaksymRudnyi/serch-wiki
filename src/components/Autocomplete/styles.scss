.input-suggestions {
    padding: 10px;
    border: 1px solid #dfe1e5;
    border-radius: 10px;
    margin-top: 10px;
    width: 100%;
    right: 0;
    left: 0;
    box-sizing: border-box;

    &-item {
        padding: 4px 0;
        text-align: left;
        box-sizing: border-box;
    }

    .search-link {
        display: block;
        margin-top: 10px;
        padding: 4px 0;
    }
}