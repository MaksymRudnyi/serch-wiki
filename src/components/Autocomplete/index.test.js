import React from 'react';
import { shallow } from 'enzyme'

import Autocomplete from '.';

const render = (props) => shallow(<Autocomplete {...props}/>);

describe('Autocomplete component', () => {
    let sut;
    let props;

    describe('without props', () => {
        beforeEach(() => {
            sut = render(props);
        });

        it('should match snapshot', () => {
            expect(sut).toMatchSnapshot();
        });
    });

    describe('when articles are passed', () => {
        beforeEach(() => {
            props = {
                articles: [1],
                searchValue: 'search value',
                onSearchChange: jest.fn()
            };
            sut = render(props);
        });

        it('should match renderMenu snapshot', () => {
            const props = sut.props();

            expect(props.renderMenu()).toMatchSnapshot();
        });
    });

    describe('when articles are NOT passed', () => {
        beforeEach(() => {
            props = {
                articles: null,
                searchValue: 'search value',
                onSearchChange: jest.fn()
            };
            sut = render(props);
        });

        it('should match renderMenu snapshot', () => {
            const props = sut.props();

            expect(props.renderMenu()).toMatchSnapshot();
        });
    });

    describe('when renderItem props is called', () => {
        beforeEach(() => {
            sut = render(props);
        });

        describe('when item is highlighted', () => {
            it('should match renderMenu snapshot', () => {
                const { renderItem } = sut.props();
                const item = {
                    id: 1,
                    label: 'test'
                };

                expect(renderItem(item, true)).toMatchSnapshot();
            });
        });

        describe('when item is NOT highlighted', () => {
            it('should match renderMenu snapshot', () => {
                const { renderItem } = sut.props();
                const item = {
                    id: 1,
                    label: 'test'
                };

                expect(renderItem(item, false)).toMatchSnapshot();
            });
        });
    });

    describe('when getItemValue is called', () => {
        beforeEach(() => {
            sut = render(props);
        });

        it('should return label', () => {
            const { getItemValue } = sut.props();

            expect(getItemValue({label: 'test'})).toBe('test')
        })
    })
});