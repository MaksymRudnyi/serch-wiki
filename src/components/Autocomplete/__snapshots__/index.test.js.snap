// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Autocomplete component when articles are NOT passed should match renderMenu snapshot 1`] = `<React.Fragment />`;

exports[`Autocomplete component when articles are passed should match renderMenu snapshot 1`] = `
<div
  className="input-suggestions"
  style={Object {}}
>
  <a
    className="search-link"
    href="/search?query=undefined"
  >
    See all results
  </a>
</div>
`;

exports[`Autocomplete component when renderItem props is called when item is NOT highlighted should match renderMenu snapshot 1`] = `
<div
  className="input-suggestions-item"
  style={
    Object {
      "backgroundColor": "transparent",
    }
  }
>
  <a
    href={1}
    target="_blank"
  >
    test
  </a>
</div>
`;

exports[`Autocomplete component when renderItem props is called when item is highlighted should match renderMenu snapshot 1`] = `
<div
  className="input-suggestions-item"
  style={
    Object {
      "backgroundColor": "#eee",
    }
  }
>
  <a
    href={1}
    target="_blank"
  >
    test
  </a>
</div>
`;

exports[`Autocomplete component without props should match snapshot 1`] = `
<Autocomplete
  autoHighlight={true}
  getItemValue={[Function]}
  inputProps={
    Object {
      "placeholder": "Input a search term: ",
    }
  }
  isItemSelectable={[Function]}
  menuStyle={
    Object {
      "background": "rgba(255, 255, 255, 0.9)",
      "borderRadius": "3px",
      "boxShadow": "0 2px 12px rgba(0, 0, 0, 0.1)",
      "fontSize": "90%",
      "maxHeight": "50%",
      "overflow": "auto",
      "padding": "2px 0",
      "position": "fixed",
    }
  }
  onChange={[Function]}
  onMenuVisibilityChange={[Function]}
  onSelect={[Function]}
  renderInput={[Function]}
  renderItem={[Function]}
  renderMenu={[Function]}
  selectOnBlur={false}
  slassName="autocomplete"
  value=""
  wrapperProps={Object {}}
  wrapperStyle={
    Object {
      "display": "inline-block",
    }
  }
/>
`;
